src/database.c

```diff
@@ -395,7 +405,35 @@ _add_to_db(ProfMessage* message, char* type, const Jid* const from_jid, const Ji
     if (!type) {
         type = (char*)_get_message_type_str(message->type);
     }
+    
+    // Check last-message-correction validity
+    if(message->replace_id){
+        auto_sqlite char* replace_check_query = sqlite3_mprintf("SELECT `from_jid` FROM `ChatLogs` WHERE `stanza_id` = '%q'",
+                                                                message->replace_id ? message->replace_id : "");
+
+        if (!replace_check_query) {
+            log_error("log_database_add(): SQL query for LMC check. could not allocate memory");
+            return;
+        }
+
+        sqlite3_stmt* lmc_stmt = NULL;
+
+        if (SQLITE_OK == sqlite3_prepare_v2(g_chatlog_database, replace_check_query, -1, &lmc_stmt, NULL)) {
+            if (sqlite3_step(lmc_stmt) == SQLITE_ROW) {
+                const char* from_jid_orig = (const char*)sqlite3_column_text(lmc_stmt, 0);
+
+                if (g_strcmp0(from_jid_orig, from_jid->barejid) != 0) {
+                    log_error("log_database_add(): Mismatch between 'from_jid' in the database and the current message for LMC. Corrected message sender: %s; original message sender: %s; replace_id: %s; message: %s", from_jid->barejid, from_jid_orig, message->replace_id, message->plain);
+                    cons_show_error("%s sent message correction with invalid ID. See log for details.", from_jid->barejid);
```

Comment:
I believe this error message is wrong. It's not about "invalid ID" but not matching sender.